src/v1.js

```diff
@@ -133,19 +135,36 @@ api.declare({
   ].join('\n'),
 }, async function(req, res) {
   let prefix = req.query.prefix;
-
-  // Load all clients
-  // TODO: as we acquire more clients, perform the prefix filtering in Azure
-  let clients = [];
-  await this.Client.scan({}, {
-    handler: client => {
+  let continuationToken  = req.query.continuationToken || undefined;
+  let limit         = parseInt(req.query.limit || 10, 10);
+  let Client = this.Client;
+  let resolver = this.resolver;
+
+  const method = async function(limit, continuationToken) {
+    let response = {clients: [], continuationToken};
+    let data = await Client.scan({}, {limit, continuation: continuationToken});
+    data.entries.forEach(client => {
       if (!prefix || client.clientId.startsWith(prefix)) {
-        clients.push(client.json(this.resolver));
+        response.clients.push(client.json(resolver));
       }
-    },
-  });
+    });
+    response.continuationToken = data.continuation;
+    return response;
+  };
+
+  const retryMethod = async function(limit, continuationToken, retries) {
+    let i = 1;
+    let response = await method(limit, continuationToken);
+    while (i <= retries && _.isEmpty(response.clients)) {
+      continuationToken = response.continuationToken;
+      response = await method(limit, continuationToken);
+      i++;
+    }
+    return response;
+  };
 
-  res.reply(clients);
+  let response = await retryMethod(limit, continuationToken, 1);
+  res.reply(response);
```

Comment:
I think this could be implemented more simply with a `while` loop, instead of two nested functions.  Something like

```
while (true) {
  scan with continuationToken / limit
  if results
    return them
  else
    update continuationToken
}
```

Do you want to try refactoring along those lines?  The nice thing about having tests written is, you'll know immediately if your refactor works or not :)

Reply:
@djmitche refactored the code as you suggested.